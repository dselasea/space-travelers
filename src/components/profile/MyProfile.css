.profile-page {
  max-width: 1000px;
}
