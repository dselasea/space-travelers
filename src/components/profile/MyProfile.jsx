import React from 'react';
import { useSelector } from 'react-redux';
import ProfileData from './ProfileData';
import './MyProfile.css';

const MyProfile = () => {
  const { rockets } = useSelector((state) => state.rockets);
  const { missions } = useSelector((state) => state.missions);
  return (
    <div className="p-5 m-auto profile-page">
      <div className="row">
        <ProfileData missions={missions} rockets={rockets} />
      </div>
    </div>
  );
};

export default MyProfile;
