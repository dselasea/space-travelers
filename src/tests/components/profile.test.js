import { screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import MyProfile from '../../components/profile/MyProfile';
import renderWithProviders from './utils';

describe('Profile page', () => {
  beforeEach(() => {
    renderWithProviders(
      <MemoryRouter>
        <MyProfile />
      </MemoryRouter>,
    );
  });

  it('renders correctly', () => {
    const { profile } = renderWithProviders(
      <MemoryRouter>
        <MyProfile />
      </MemoryRouter>,
    );
    expect(profile).toMatchSnapshot();
  });

  it('headers are present in the document', () => {
    const missionsHeader = screen.getByText('My Missions');
    const rocketsHeader = screen.getByText('My Rockets');

    expect(missionsHeader).toBeInTheDocument();
    expect(rocketsHeader).toBeInTheDocument();
  });
});
