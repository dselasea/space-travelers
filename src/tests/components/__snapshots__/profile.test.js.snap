// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page renders correctly 1`] = `undefined`;
